lib/services/sqs.js

```diff
@@ -118,7 +118,8 @@ AWS.util.update(AWS.SQS.prototype, {
    * @api private
    */
   buildEndpoint: function buildEndpoint(request) {
-    var url = request.httpRequest.params.QueueUrl;
+    var params = request.httpRequest.params || this.config.params;
+    var url = params.QueueUrl;
```

Comment:
in JSON protocol, request.httpRequest.params no longer contained any QueueUrl in testing when the client was initialized with

```js
new AWS.SQS({
  params: { QueueUrl: ... }
});
```

Reply:
The change on this line is needed if this is a way that users pass QueueUrl to client operations, and not merely a testing mechanism.

Reply:
After investigation, I believe this change is good to go as-is. 

The query protocol assigns client init params to each httpRequest but the JSON protocol does not. Rather than modify the protocols behaviors, modifying the behavior of the SQS customizations is the better option for limiting potential impact.